src/framework/ui/dropdown/dropdown.component.tsx

```diff
@@ -0,0 +1,751 @@
+        animationStyle={animatedStyle}
+      />
+    );
+  };
+
+  private renderIconElement = (style: ImageStyle): IconElement => {
+    const { icon } = this.props;
+    const { visible } = this.state;
+
+    if (icon) {
+      return icon(style, visible);
+    } else {
+      return this.renderDefaultIconElement(style);
+    }
+  };
+
+  private renderTextElement = (valueStyle: TextStyle, placeholderStyle: TextStyle): TextElement => {
+    const { placeholder, textStyle } = this.props;
+    const value: string = this.strategy.getPlaceholder(placeholder);
+    const style: TextStyle = placeholder === value ? placeholderStyle : valueStyle;
+
+    return (
+      <Text
+        style={[style, styles.text, textStyle]}
+        numberOfLines={1}
+        ellipsizeMode='tail'>
+        {value}
+      </Text>
+    );
+  };
+
+  private renderMenuElement = (style: StyleType): DropdownMenuElement => {
+    const { appearance, selectedOption, ...restProps } = this.props;
+    const additionalMenuStyle: StyleType = { width: this.state.menuWidth };
+
+    return (
+      <DropdownMenu
+        {...restProps}
+        strategy={this.strategy}
+        key={0}
+        style={[styles.menu, style, additionalMenuStyle]}
+        bounces={false}
+        onSelect={this.onItemSelect}
+      />
+    );
+  };
+
+  private renderControlChildren = (style: StyleType): [IconElement, TextElement] => {
+    return [
+      this.renderIconElement(style.icon),
+      this.renderTextElement(style.text, style.placeholder),
+    ];
+  };
+
+  private renderControlElement = (): ControlElement => {
+    const { themedStyle, controlStyle, ...restProps } = this.props;
+    const { control, outline, ...childrenStyles } = this.getComponentStyle(themedStyle);
+    const [iconElement, textElement] = this.renderControlChildren(childrenStyles);
+
+    const measuringProps: MeasuringElementProps = { tag: MEASURED_CONTROL_TAG };
+
+    return (
+      <MeasureNode onResult={this.onControlMeasure}>
+        {[
+          <TouchableOpacity
+            {...restProps}
+            {...measuringProps}
+            key={MEASURED_CONTROL_TAG}
+            activeOpacity={1.0}
+            style={[styles.control, control, controlStyle]}
+            onPress={this.onPress}
+            onPressIn={this.onPressIn}
+            onPressOut={this.onPressOut}>
+            {textElement}
+            {iconElement}
+          </TouchableOpacity>,
+        ]}
+      </MeasureNode>
+    );
+  };
+
+  private renderComponentChildren = (style: StyleType): DropdownChildren => {
+    const { label } = this.props;
+
+    return [
+      this.renderMenuElement(style.menu),
+      isValidString(label) && this.renderLabelElement(style.label),
+      this.renderControlElement(),
+    ];
+  };
+
+  public render(): DropdownElement {
+    const { themedStyle, style } = this.props;
+    const { visible, menuWidth } = this.state;
+    const evaStyles: StyleType = this.getComponentStyle(themedStyle);
```

Comment:
let's use our common naming for this variable

Reply:
done